--- gunicorn/workers/base.py.orig	2019-02-12 17:56:24.010313733 +0100
+++ gunicorn/workers/base.py	2019-02-12 17:56:39.694313987 +0100
@@ -185,10 +185,15 @@

     def handle_quit(self, sig, frame):
         self.alive = False
-        # worker_int callback
-        self.cfg.worker_int(self)
-        time.sleep(0.1)
-        sys.exit(0)
+        try:
+            self.app.zato_wsgi_app.cleanup_on_stop()
+        except Exception as e:
+            # At this poing logging may not be available anymore hence we are using print() instead.
+            from traceback import format_exc
+            print('Exception in gunicorn.workers.base.Worker.handle_quit', format_exc())
+        finally:
+            time.sleep(0.1)
+            sys.exit(0)

     def handle_abort(self, sig, frame):
         self.alive = False
